ucca/constructions.py

```diff
@@ -70,7 +71,8 @@ def __init__(self, edge, reference=None, reference_yield_tags=None, verbose=Fals
         self.verbose = verbose
         self.terminals = self.edge.child.get_terminals()
         self._terminal_yield = positions(self.terminals)
-        self._terminal_yield_no_punct = positions(self.edge.child.get_terminals(punct=False))
+        self._terminal_yield_no_punct = positions(self.edge.parent.get_terminals(punct=False)) if self.is_implicit\
```

Comment:
Good. Just had an idea: it can be rewritten `positions((self.edge.parent if self.is_implicit else self.edge.child).get_terminals(punct=False))`